dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx

```diff
@@ -142,30 +148,49 @@ export const DataConfigPanelItem = ({ fieldOptionList, visualizations }: any) =>
     setConfigList(updatedList);
   };
 
+  const updateCoordinateConfig = (
+    value: string,
+    type: string,
+    field: string,
+    dataType?: string
+  ) => {
+    const list = { ...configList };
+    let listItem = { ...list[type][0] };
+    listItem[field] = value;
+    if (field === 'label') {
+      listItem.name = value;
+      listItem.type = dataType !== undefined ? dataType : '';
+    }
+    const updatedList = {
+      ...list,
```

Comment:
You do not need to spread the object here. This is a local variable, that was not updated after initialization. You can return the variable as is. If the calling function need to use the variable, it may use spread operator to create a new object.